src/zdaemon/tests/tests.py

```diff
@@ -79,7 +79,7 @@ def dont_hang_when_program_doesnt_start():
     >>> write('conf',
     ... '''
     ... <runner>
-    ...   program sleep
+    ...   program sleep .1
```

Comment:
The PEP suggests two possible fixes:

1. register a custom SIGCHLD handler that raises a custom exception
2. use signal.set_wakeup_fd() and add it to the `r` set we're passing to `select()`

I like option 1 better